### PR TITLE
Nisha | Test | Convert cmd+k and help side bar in material

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/85ztj3nvh
https://app.clickup.com/t/85ztj3nyt

### CODE REVIEW

File: apps/web-giddh/src/app/shared/header/header.component.html [hunk 1]
```diff
@@ -505,13 +505,18 @@ <h3 class="modal-title pull-left">{{ localeData?.ready_to_upgrade }}</h3>
...
         class="help-support-aside"
         *ngIf="asideHelpSupportMenuState === 'in'"
         (clickOutside)="closeHelpPaneOnOutsideClick()"
     >
         <aside-help-support (closeAsideEvent)="toggleHelpSupportPane(false)"></aside-help-support>
-    </div>
+    </div> -->
+
+    <ng-template #asideHelpSupportMenuStateRef>
+        <aside-help-support (closeAsideEvent)="matDialogRef.close()"></aside-help-support>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.component.html]:
```suggestion
        <aside-help-support></aside-help-support>
```

File: apps/web-giddh/src/app/shared/header/components/aside-help-support/aside-help-support.component.html [hunk 2]
```diff
@@ -6,93 +6,103 @@
...
-                        <a href="https://giddh.canny.io/feature-requests" target="_blank">{{
-                            localeData?.feature_request
-                        }}</a>
-                    </li>
-                    <li>
-                        <a href="https://changelog.giddh.com" target="_blank">{{ localeData?.changelog }}</a>
-                    </li>
-                </ul>
+            <h2 mat-dialog-title>{{ localeData?.page_heading }}</h2>
+            <div mat-dialog-close id="close" class="cursor-pointer" (click)="closeAsidePane($event)">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/aside-help-support/aside-help-support.component.html]:
```suggestion
            <div mat-dialog-close id="close" class="cursor-pointer">
```

Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/components/aside-help-support/aside-help-support.component.html]: delete closeAsidePane method from .ts file also if it's unused

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 3]
```diff
@@ -220,6 +222,7 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
     public isCalendlyModelActivate: boolean = false;
     /** Calendly url */
     public calendlyUrl: any = '';
+    public matDialogRef: any;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.component.ts]: remove

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 4]
```diff
@@ -831,18 +836,17 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
      * @param {boolean} show
      * @memberof HeaderComponent
      */
-    public toggleHelpSupportPane(show: boolean): void {
-        setTimeout(() => {
-            if (show) {
-                this.asideSettingMenuState = 'out';
-                document.querySelector('body')?.classList?.remove('aside-setting');
+    public toggleHelpSupportPane(): void {
+        this.matDialogRef = this.dialog.open(this.asideHelpSupportMenuStateRef,{
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.component.ts]:
```suggestion
        this.dialog.open(this.asideHelpSupportMenuStateRef,{
```

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 5]
```diff
@@ -1396,7 +1400,7 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
      * @memberof HeaderComponent
      */
     public addClassInBodyIfPageHasTabs(): void {
-        this.toggleHelpSupportPane(false);
+        // this.toggleHelpSupportPane(false);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.component.ts]: why is this commented?

File: apps/web-giddh/src/app/shared/header/header.module.ts [hunk 6]
```diff
@@ -49,7 +51,11 @@ import { HeaderComponent } from "./header.component";
         RouterModule,
         MatMenuModule,
         MatButtonModule,
-        MatTreeModule
+        MatTreeModule,
+        MatDialogModule,
+        MatDialogModule,
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.module.ts]: remove duplicate import

File: apps/web-giddh/src/app/theme/command-k/command.k.component.html [hunk 7]
```diff
@@ -151,7 +154,7 @@
                     (paste)="onPasteInSearch()"
                 />
             </div>
-            <i class="icon-cross close-navigation" (click)="close()"></i>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/theme/command-k/command.k.component.html:154]: remove close method from .ts file if its unused

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 8]
```diff
@@ -16,9 +16,9 @@ $custom-typography: mat-typography-config(
 @import "../css/global/custom-fields.scss";
 @import "../css/global/inventory.scss";
 @import "../css/global/all-items.scss";
-@import "../css/global/header.scss";
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/css/style-2.scss:19]: restrict this from getting deleted

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/header/components/aside-help-support/aside-help-support.component.ts [hunk 9]
```diff
@@ -65,7 +55,6 @@ export class AsideHelpSupportComponent implements OnInit, OnDestroy {
      */
     public scheduleNow(event): void {
         this.store.dispatch(this.generalActions.isOpenCalendlyModel(true));
-        this.closeAsidePane(event);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/aside-help-support/aside-help-support.component.ts]: Close dialog when click on schedule

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 10]
```diff
@@ -92,6 +93,7 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
     @ViewChild('allModulesPopover', { static: true }) public allModulesPopover: PopoverDirective;
     /** Instance of menu trigger */
     @ViewChild(MatMenuTrigger) public trigger: MatMenuTrigger;
+    @ViewChild('asideHelpSupportMenuStateRef', { static: true }) public asideHelpSupportMenuStateRef: TemplateRef<any>;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/header.component.ts:97]: Add documentation for instance

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 11]
```diff
@@ -258,7 +260,9 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
         public location: Location,
         private localeService: LocaleService,
         private settingsFinancialYearActions: SettingsFinancialYearActions,
-        private sanitizer: DomSanitizer
+        private sanitizer: DomSanitizer,
+        public dialog: MatDialog,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/header.component.ts]: remove last comma

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 12]
```diff
@@ -831,18 +835,21 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
...
-                this.asideSettingMenuState = 'out';
-                document.querySelector('body')?.classList?.remove('aside-setting');
-            }
-            this.asideInventorySidebarMenuState = 'out'
-            document.querySelector('body').classList.remove('mobile-setting-sidebar');
-            this.asideHelpSupportMenuState = (show && this.asideHelpSupportMenuState === 'out') ? 'in' : 'out';
-            this.toggleBodyClass();
-        }, (this.asideHelpSupportMenuState === 'out') ? 100 : 0);
+    public toggleHelpSupportPane(event: boolean): void {
+        if(event){
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/header.component.ts]: Add documentation for this function and format code

File: apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts [hunk 13]
```diff
@@ -122,7 +123,8 @@ export class PrimarySidebarComponent implements OnInit, OnChanges, OnDestroy {
         private dbService: DbService,
         private groupWithAction: GroupWithAccountsAction,
         private localeService: LocaleService,
-        private salesAction: SalesActions
+        private salesAction: SalesActions,
+        public dialog: MatDialog,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts:127]: remove last comma

File: apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts [hunk 14]
```diff
@@ -177,7 +179,10 @@ export class PrimarySidebarComponent implements OnInit, OnChanges, OnDestroy {
             event.preventDefault();
             event.stopPropagation();
             if (this.companyList?.length > 0) {
-                this.showNavigationModal();
+                this.dialog.open(this.navigationModal, {
+                    width:'630px',
+                    height: '600',
+                })
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts]: remove the last comma and add semicolon on the end dialog function

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Review by dilpreetsinghofficial — APPROVED** [review]

**Pull request merged**